### PR TITLE
Centralize Paths (Possibly fixes #43)

### LINKED ISSUE #43: Traceback (guilds not appearing)
## Expected Behavior
Application starts
...

## Actual Behavior
See logs and notes
...

## Steps to reproduce the problem

  1. Download docker image
  2. Attempt to run
  3. (See notes)

### Environment

  - Operating system: Linux (see comments
  - CPU Architecture: x86_64 (PC)
  - Flavor: Docker

### DiscordChatExporter-frontend

  - [Release](https://github.com/slatinsky/DiscordChatExporter-frontend/releases) version or commit hash: docker hash -  c31fe828b048 (most recent main)
  - Can the issue be reproduced in the [demo](https://dcef.slada.sk/)?: Don't know

### [DiscordChatExporter](https://github.com/Tyrrrz/DiscordChatExporter)

- Version (CLI `--version`): v2.42.2

How did you export the data?
  - Export format (CLI `--format`): Json
  - Downloaded assets (CLI `--media`): Yes (and --reuse-media_
  - Markdown processing (CLI `--markdown`): set to false
  - Other CLI options (optional): ...

  

## Logs

Is there any relevant information in logs? If so, please provide them:

```
2023-11-05 04:49:44.915173+00:00[Preprocess] took 0.64 seconds
2023-11-05 04:49:44.915217+00:00Traceback (most recent call last):
2023-11-05 04:49:44.915228+00:00File "/dcef/backend/preprocess/main_mongo.py", line 146, in <module>
2023-11-05 04:49:44.915368+00:00main(input_dir, output_dir)
2023-11-05 04:49:44.915413+00:00File "/dcef/backend/preprocess/main_mongo.py", line 119, in main
2023-11-05 04:49:44.915485+00:00channel_cache = ChannelCache()
2023-11-05 04:49:44.915507+00:00^^^^^^^^^^^^^^
2023-11-05 04:49:44.915521+00:00File "/dcef/backend/preprocess/ChannelCache.py", line 22, in __init__
2023-11-05 04:49:44.915568+00:00os.makedirs(self.cache_folder_path)
2023-11-05 04:49:44.915594+00:00File "<frozen os>", line 215, in makedirs
2023-11-05 04:49:44.915605+00:00File "<frozen os>", line 215, in makedirs
2023-11-05 04:49:44.915617+00:00File "<frozen os>", line 215, in makedirs
2023-11-05 04:49:44.915624+00:00[Previous line repeated 1 more time]
2023-11-05 04:49:44.915630+00:00File "<frozen os>", line 225, in makedirs
2023-11-05 04:49:44.915636+00:00OSError: [Errno 30] Read-only file system: '../../release'
```

**Notes**

This is actually running on a TrueNAS SCALE system, which is a debian linux appliance that runs a kubernetes cluster. I've ran into a few weird readonly errors prior to this but was able to resolve them. After I saw the most recent commit was to fix the docker image, I also made sure to pull the image by sha256 digest to make sure it was the most up to date.

`/dcef/cache` -> PVC Volume
`/dcef/exports` -> points to my DCE exports directory. Exports were generated with incremental chat exporter.
`/var/lib/nginx` -> PVC Volume (without this, /var/lib/nginx/body can't be written, NGINX throws an EMERG error and the container quits)
`/var/log/nginx` -> PVC Volume (without this, being unable to write access.log triggers an EMERG error and container quits)

The container is running as uid/gid 0/0 (root) as without this an error is thrown. Ports are mapped 21011 to 21011, which then feeds into a kubernetes ingress controller. Access works fine, webpage loads fine, it's only loading the guilds that fails.

I know this is probably beyond what was intended, so if there's anyway I can help let me know.

### PR DESCRIPTION
**Pull request by zyradyl**
Hello!

I didn't want to just complain so I went through and made some quick changes to the paths. I don't have a dockerhub account and I'm not really sure how to build docker images in the first place so I can't test it, but they're relatively simple changes.

Hope this helps! Thanks for the awesome software!

### CODE REVIEW

File: backend/nginx/conf/nginx-docker.conf [hunk 1]
```diff
@@ -20,6 +20,8 @@ events {
 http {
     include       mime.types;
     default_type  application/octet-stream;
+    
+    client_body_temp_path /dcef/cache/nginx/body
```
Comment (slatinsky) [inline backend/nginx/conf/nginx-docker.conf:24]: I think that `/tmp/` directory is better for this purpose. It is temporary data, not meant to be persistent. Also missing required `;` at the end.

For example:
```conf
client_body_temp_path /tmp/nginx_client_body 1 2;
```

File: backend/nginx/conf/nginx-docker.conf [hunk 2]
```diff
@@ -35,6 +37,7 @@ http {
     server {
         listen       21011;
         server_name  localhost;
+        access_log   /dcef/cache/nginx/logs/access.log
```
Comment (slatinsky) [inline backend/nginx/conf/nginx-docker.conf:40]: You may want to reroute error_log too. Also missing required ; at the end.

for example:
```conf
access_log   /dcef/cache/nginx/logs/access.log;
error_log    /dcef/cache/nginx/logs/error.log;
```

This change also need /dcef/cache/nginx/logs/ directory to exists, else nginx will complain

File: backend/preprocess/ChannelCache.py [hunk 3]
```diff
@@ -16,7 +16,7 @@ def __init__(self):
 		if is_compiled():
 			self.cache_folder_path = "../../storage/cache/message-ids"
 		else:
-			self.cache_folder_path = "../../release/dcef/storage/cache/message-ids"
+			self.cache_folder_path = "../../cache/message-ids"
```
Comment (slatinsky) [inline backend/preprocess/ChannelCache.py:19]: This change will break development environment. I will restructure the project folder structure to fix paths later

File: backend/nginx/conf/nginx-docker.conf (hunk 1 above)
Comment (zyradyl) [reply backend/nginx/conf/nginx-docker.conf]: Force pushed with your recommended change, Thanks!

File: backend/nginx/conf/nginx-docker.conf (hunk 2 above)
Comment (zyradyl) [reply backend/nginx/conf/nginx-docker.conf]: Force pushed with your recommended change, Thanks! Also added a directory make line to `run_container.sh`.

File: backend/preprocess/ChannelCache.py (hunk 3 above)
Comment (zyradyl) [reply backend/preprocess/ChannelCache.py:19]: Is there anything I can do to resolve this to make it easier on you? I am on my way to bed now as its 11PM here, but after I get back from school tomorrow I can make the changes. Where would I look to find the breakage?

**Comment by slatinsky** [comment]
Hello,

you don't need dockerhub to build the docker image. Just follow instructions in [README.md#other-ways-to-run-dcef](https://github.com/slatinsky/DiscordChatExporter-frontend#other-ways-to-run-dcef) and build docker image from source. - TL;DR `docker rm dcef -f && docker build -t dcef . && docker run --restart=always --name dcef -p 21011:21011 -it dcef`

Also it is expected that you would add your own reverse proxy (for example nginx) to handle logging, SSL certificates, rate limiting,... . That's why nginx logs were not made to be persistent in `/dcef/cache`

Sorry, I won't merge this pull request as is, project needs bigger folder restructure to fix this issue (moving `backend` and `frontend` folders to `src` folder). But thanks for your contribution <3

**Pull request closed without merging**